labman/gui/handlers/process_handlers/sample_plating_process.py

```diff
@@ -77,12 +77,21 @@ def sample_plating_process_handler_patch_request(
             study_id = int(req_path[3])
             well_attribute = req_path[4]
 
-            blank_or_unknown = False
-            try:
-                sample_id = Study(study_id).specimen_id_to_sample_id(req_value)
-            except ValueError:
-                sample_id = req_value
-                blank_or_unknown = True
+            sample_id = req_value
+            blank_or_unknown = True
+
+            # It actually IS possible to plate a plate without specifying
+            # separate study id(s); can plate just all blanks, or can provide
+            # the fully qualified sample id(s)--i.e., <studyid>.<sampleid>.
+            if study_id != 0:
+                try:
+                    sample_id = Study(study_id).specimen_id_to_sample_id(
+                        req_value)
+                    blank_or_unknown = False
```

Comment:
Should the setting of this variable be done within `except`?

Reply:
Counterintuitively, no :)  The default values of the variables `sample_id` and `blank_or_unknown` is now done before the `try`, and these are the values that are used if either (a) `study_id` is not 0 OR (b) the `try` fails with a `ValueError` (see * below).  If the `try` fails with anything else then the entire function bails, so it doesn't matter what those variables are.  If and only if the `try` succeeds are the variables set to the values within the `try`.


* Note that the `try` fails iff the `Study(study_id).specimen_id_to_sample_id()` call fails, as the `blank_or_unknown = False` can't really fail ... this assures that we can't realistically end up in an inconsistent situation where `sample_id` has been set during the `try` but `blank_or_unknown` has *not*.  Thus, the ordering of the two statements within the `try` is really important.

Reply:
ahhhh, okay, thank you for the explanation!